python-package/xgboost/core.py

```diff
@@ -715,9 +737,7 @@ def set_group(self, group):
         group : array like
             Group size of each group
         """
-        _check_call(_LIB.XGDMatrixSetGroup(self.handle,
-                                           c_array(ctypes.c_uint, group),
-                                           c_bst_ulong(len(group))))
+        self.set_uint_info('group', group)
```

Comment:
Are you sure this is equivalent to `XGDMatrixSetGroup()`?

Reply:
Good point.  `SetInfo` does a copy first, while `SetGroup` builds the `group_ptr` on the fly, which should be slightly faster.  But other than this, the code is duplicated.  It's a little bit tricky to add a test, since we can't get the group back without another function transforming `group_ptr` into the original `group`.  We can postpone it to #4150.  But I am happy to add such test if you think it's better.